src/Nethermind/Nethermind.Shutter.Test/ShutterBlockHandlerTests.cs

```diff
@@ -0,0 +1,98 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using NUnit.Framework;
+using Nethermind.Core.Test.Builders;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.Core;
+using System;
+
+namespace Nethermind.Shutter.Test;
+
+[TestFixture]
+class ShutterBlockHandlerTests
+{
+    [Test]
+    public async Task Can_wait_for_valid_block()
+    {
+        Random rnd = new(ShutterTestsCommon.Seed);
+        Timestamper timestamper = ShutterTestsCommon.InitTimestamper(ShutterTestsCommon.InitialSlotTimestamp, 0);
+
+        ShutterApiSimulator api = ShutterTestsCommon.InitApi(rnd, timestamper);
+        IShutterBlockHandler blockHandler = api.BlockHandler;
+
+        bool waitReturned = false;
+        CancellationTokenSource source = new();
+        _ = blockHandler.WaitForBlockInSlot(ShutterTestsCommon.InitialSlot, source.Token)
+            .ContinueWith((_) => waitReturned = true)
+            .WaitAsync(source.Token);
+
+        await Task.Delay((int)(api.BlockWaitCutoff.TotalMilliseconds / 2));
```

Comment:
making tests that are dependent on time often results in them being unstable in CI. 
Its possible to inject a custom implementation of `Task.Delay` or `CancellationTokenSource`, that can be used in tests to control exact timeouts or returning results.